### DISCUSSION TITLE
[Idea] Add a parameter to be able to set debug level

### DISCUSSION THREAD

**Discussion by web-cooking-factory** [PrestaShop/PrestaShop#31606] [Ideas]
### The problem

Sometimes we only want errors to be raised and not notices or warnings. We should be able to set the debug mode level.

### The solution

Add a parameter in BO.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [x] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
I'm afraid I have to disagree, I mean, I don't like this idea. Notices may lead to unexpected behaviors and should be treated as bugs in the code to be fixed.

**Reply by web-cooking-factory** [reply]
Yes, in a perfect world :)
In the real world, having Notice: undefined index raised by third party modules drives me crazy when I'm tracking an PHP fatal error.
The debug level is a parameter of Symfony Debug::enable(). We already have the _PS_DISPLAY_COMPATIBILITY_WARNING_ constant, I would like to have _PS_DISPLAY_ONLY_ERRORS_.
As a developper, I should have the choice.

**Reply by kpodemski** [reply]
Hmm, it could be the setting that is only available if you are in the developer mode already. What do you think?

**Reply by web-cooking-factory** [reply]
Yes why not, as long as it exists !
